lib/rust/ensogl/component/grid-view/src/basic.rs

```diff
@@ -0,0 +1,182 @@
+        display_object.add_child(&background);
+        if let Some(layer) = text_layer {
+            label.add_to_scene_layer(layer);
+        }
+        Self { display_object, label, background }
+    }
+
+    fn update_layout(
+        &self,
+        entry_size: Vector2,
+        bg_margin: f32,
+        text_size: text::Size,
+        text_offset: f32,
+    ) {
+        use entry_background::PADDING_PX;
+        let bg_size = entry_size - Vector2(bg_margin, bg_margin) * 2.0;
+        let bg_size_with_padding = bg_size + Vector2(PADDING_PX, PADDING_PX) * 2.0;
+        self.background.size.set(bg_size_with_padding);
+        self.label.set_position_xy(Vector2(text_offset - entry_size.x / 2.0, text_size.raw / 2.0));
+    }
+}
+
+
+// === Entry ===
+
+/// A [`BasicGridView`] entry - a label with background.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Entry {
+    frp:  EntryFrp<Self>,
+    data: Rc<EntryData>,
+}
+
+impl crate::Entry for Entry {
+    type Model = ImString;
+    type Params = EntryParams;
+
+    fn new(app: &Application, text_layer: &Option<Layer>) -> Self {
+        let data = Rc::new(EntryData::new(app, text_layer));
+        let frp = EntryFrp::<Self>::new();
+        let input = &frp.private().input;
+        let network = frp.network();
+
+        enso_frp::extend! { network
+            bg_color <- input.set_params.map(|params| params.bg_color).on_change();
+            bg_margin <- input.set_params.map(|params| params.bg_margin).on_change();
+            font <- input.set_params.map(|params| params.font.clone_ref()).on_change();
+            text_offset <- input.set_params.map(|params| params.text_offset).on_change();
+            text_color <- input.set_params.map(|params| params.text_color).on_change();
+            text_size <- input.set_params.map(|params| params.text_size).on_change();
+
+            layout <- all(input.set_size, bg_margin, text_size, text_offset);
+            eval layout ((&(es, m, ts, to)) data.update_layout(es, m, ts, to));
+
+            eval bg_color ((color) data.background.color.set(color.into()));
+            data.label.set_default_color <+ text_color.on_change();
+            data.label.set_font <+ font.on_change().map(ToString::to_string);
+            data.label.set_default_text_size <+ text_size.on_change();
+
+            content <- input.set_model.map(|s| s.to_string());
+            max_width_px <- input.set_size.map(|size| size.x);
+            data.label.set_content_truncated <+ all(&content, &max_width_px);
+        }
+        Self { frp, data }
+    }
+
+    fn frp(&self) -> &EntryFrp<Self> {
+        &self.frp
+    }
+}
+
+impl display::Object for Entry {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.data.display_object
+    }
+}
+
+// =====================
+// === BasicGridView ===
+// =====================
+
+/// The Basic version of Grid View, where each entry is just a label with background.
+pub type BasicGridView = crate::GridView<Entry>;
```

Comment:
I think I'd call it "simple" instead of "basic". The word "simple" is pretty common in UX frameworks, while I haven't seen "basic" ever. I think that "basic" has a connotation of something functionally limited, while this is visually limited only.